### ISSUE TITLE
Accessebility enhancements on tables/grids (A)

### ISSUE THREAD

**Issue by Febakke** [Altinn/app-frontend-react#1628]
### Description

With this issue we are breaking [WCAG 1.3.1 Info and relationships.](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)

When changing from desktop to mobile view on our tables and repeating groups the semantic structure of the table changes. We can change how it looks in the form, but the HTML needs to stay the same. 

NB: This issue is related to all our tables/grids

Screenshots of one of our tables in question
<img width="1111" alt="Skjermbilde 2023-11-02 kl  15 41 25" src="https://github.com/Altinn/app-frontend-react/assets/33222679/f4934416-5bc7-4120-a929-0d5990108e4f">
<img width="445" alt="Skjermbilde 2023-11-02 kl  15 41 35" src="https://github.com/Altinn/app-frontend-react/assets/33222679/b9536b9a-cd51-4f4e-b00e-bc528d566ee8">

### Additional Information

- Discussion about this on Slack: https://digdir.slack.com/archives/C076GU2QHCG/p1727690599903289

### Tasks
- [ ] Relevant issue: #904 
- [ ] Table deep dive: Get an overview of all our tables and describe how they behave across screen sizes
- [ ] Describe how we want to solve our accessibility issues for all our tables. We should handle this consistent between them. 

### Acceptance Criterias
- [ ] Consistent HTML table structure between view on large and smaller screens
- [ ] Input fields are connected to relevant headers/values in the table
- [x] App developer can set a caption for our tables/grids

**Comment by Febakke** [comment]
- [ ] Find example provided by SSB
- [ ] More details on how this is breaking 1.3.1

**Comment by olavflar** [comment]
https://www.accessibility-developer-guide.com/examples/tables/responsive/

**Comment by Magnusrm** [comment]
This is currently solved for "repeating groups" and "simple-table", but still remains an issue for the "grid" component.

**Comment by Ildest** [comment]
@Magnusrm lukker denne og lager ny for grid-delen.

**Comment by Magnusrm** [comment]
The remainder of this isse concering grid should be handled in this issue: #4118
